src/ucp/proto/proto_reconfig.c

```diff
@@ -34,9 +34,24 @@ static ucs_status_t ucp_proto_reconfig_progress(uct_pending_req_t *self)
 {
     ucp_request_t *req = ucs_container_of(self, ucp_request_t, send.uct);
     ucp_ep_h ep        = req->send.ep;
+    ucs_string_buffer_t strb;
 
     /* This protocol should not be selected for valid and connected endpoint */
-    ucs_assert(!(ep->flags & UCP_EP_FLAG_REMOTE_CONNECTED));
+    if (ep->flags & UCP_EP_FLAG_REMOTE_CONNECTED) {
+        ucs_string_buffer_init(&strb);
+        ucp_ep_config_name(ep->worker, req->send.proto_config->ep_cfg_index,
+                           &strb);
+        ucs_string_buffer_appendf(&strb, " | ");
```

Comment:
can you pls post an example of a full error message?
the "|" character seems weird

Reply:
`UCX  ERROR cannot find remote protocol for: 0x3b6f530 self cfg#0 | amo_fetch into host memory from rdma, arg in host memory`